tests/v2/test_waku_swap.nim

```diff
@@ -107,7 +107,7 @@ procSuite "Waku SWAP Accounting":
     var futures = [newFuture[bool](), newFuture[bool]()]
 
     # Define the waku swap Config for this test
-    let swapConfig = SwapConfig(mode: SwapMode.Mock, paymentThreshold: 1, disconnectThreshold: -1)
+    let swapConfig = SwapConfig(mode: SwapMode.Mock, paymentThreshold: 1, disconnectThreshold: -1, stateUpdateOverRide: true, closePeerConnection: false)
```

Comment:
Does this mean that we do not disconnect the peer here? What does state update override signify?

Assuming above is correct, the disconnectThreshold being set seems a bit contradictory, though I can see why. Maybe leave a comment here about how this should probably be made optional if we don't disconnect peer?

Finally, shouldn't `SwapMode.Mock` imply that (1) There is no threshold for disconnects (2) closePeerConnection is false? As opposed to setting all three things at once.

Reply:
The Mock Mode is actually meant to cover peer disconnection, unlike the Soft mode. But the reason why there was an extra boolean field indicating whether or not the disconnect should occur is because there is an uncaught error that always occurs after switch.disconnect is called. The error seems to occur in libp2p, and it causes the test cases dependent on the disconnect function to fail.